pkg/util/node/node.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"slices"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func HasTaint(node *corev1.Node, taint string) bool {
```

Comment:
Shouldn't we also check the effect?

Reply:
Good question. In this particular case it doesn't matter, but I'm wondering what's more useful in general. I'll circle back to this